File: DinnersAccounting/Domain/Entities/ContinuousProposition.cs
using System;
using System.Collections.Generic;

namespace DA.Dinners.Model
{
    /// <summary>
    /// Class for continuous proposition with start date and end date
    /// </summary>
    public class ContinuousProposition : Proposition
    {
        /// <summary>
        /// Gets or sets the start date.
        /// </summary>
        /// <value>
        /// The start date.
        /// </value>
        public DateTime StartDate { get; set; }

        /// <summary>
        /// Gets or sets the end date.
        /// </summary>
        /// <value>
        /// The end date.
        /// </value>
        public DateTime EndDate { get; set; }

        /// <summary>
        /// Gets or sets the day propositions.
        /// </summary>
        /// <value>
        /// The day propositions.
        /// </value>
        public List<DayProposition> DayPropositions { get; set; }

        public ContinuousProposition()
        {
            DayPropositions = new List<DayProposition>();
        }

        public void Init()
        {
            int days = EndDate.Subtract(StartDate).Days;
            for (int i = 0; i < days; i++)
                DayPropositions.Add(new DayProposition() { Date = StartDate.AddDays(i) });
        }
    }
}

File: DinnersAccounting/UI/Scripts/Model/ContinuousProposition.js
//-----------------------CONTINUOUS_PROPOSITION--------------------------------------------
var ContinuousProposition = function (id, dateStart, dateEnd, dayPropositions, products) {
    var self = this;
    self.Id = id;
    self.DateStart = dateStart;
    self.DateEnd = dateEnd;
    self.DayPropositions = dayPropositions;
    self.Products = products;

    return {
        Id: self.Id,
        DateStart: self.DateStart,
        DateEnd: self.DateEnd,
        DayPropositions: self.DayPropositions,
        Products: self.Products
    };
};
//-----------------------------------------------------------------------------------------